assets/apps/calendar/init.js

```diff
@@ -12,11 +12,9 @@ let handler = function (e) {
 
       mainWindow = e.source;
       origin = e.origin;
-      if (e.data.type == "ping") {
-          console.log("ping");
-      } else if (e.data.type == "load") {
+      if (e.data.type == "load") {
           initialiseCalendar(false, e.data.calendars);
-          load(e.data.previousMonth, e.data.currentMonth, e.data.nextMonth, e.data.yearMonth, e.data.username);
+          load(e.data.previousMonth, e.data.currentMonth, e.data.nextMonth, e.data.recurringEvents, e.data.yearMonth, e.data.username);
       } else if (e.data.type == "loadAdditional") {
           loadAdditional(e.data.currentMonth, e.data.yearMonth);
```

Comment:
Would it be worth splitting off all the recurrence specific things to a separate file to make it easier to see the high level init logic? recurring.js?

Reply:
Most of the recurring code is at the bottom of file. see //--rrule blocks for recurring demarcation. There is a tiny amount of shared state, which means splitting is undesirable.

Reply:
What is the shared state?

Reply:
see first //--rrule block. 4 vars